mesh/nurbs.cpp

```diff
@@ -574,6 +829,37 @@ void KnotVector::FindInterpolant(Array<Vector*> &x)
    }
 }
 
+// Routine from "The NURBS book" - 2nd ed - Piegl and Tiller
+// Algorithm A9.1 p. 369
+void KnotVector::GetInterpolant(const Vector &x, const Vector &u,
+                                Vector &a) const
+{
+   int ks;
+   real_t xi;
+
+   // Assemble collocation matrix
+   Vector shape(Order+1);
+   DenseMatrix A(NumOfControlPoints,NumOfControlPoints);
+   A = 0.0;
+
+   for (int i = 0; i < NumOfControlPoints; i++)
+   {
+      ks = GetSpan(u[i]);
+      xi = GetIp(u[i], ks);
+      CalcShape ( shape, ks-Order, xi);
+
+      for (int p = 0; p < Order+1; p++)
+      {
+         A(i, ks - Order + p ) =  shape[p];
+      }
+   }
+
+   // Solve problems
```

Comment:
Comment that this matrix is banded, and a banded solver could be used in the future for optimization?